supermarq-benchmarks/supermarq/qcvv/xeb.py

```diff
@@ -247,21 +248,27 @@ def _process_probabilities(
         Returns:
             A data frame of the full results needed to analyse the experiment.
         """
-
+        samples = list(samples)
         for sample in samples:
-            sample.sample_probabilities = sample.probabilities
-            sample.probabilities = {}
+            if sample.probabilities is None:
+                warnings.warn("Sample missing `probabilities`. This sample has been omitted.")
+                samples.remove(sample)
+            else:
+                sample.sample_probabilities = sample.probabilities
+                sample.probabilities = {}
 
         for sample in tqdm.notebook.tqdm(samples, desc="Evaluating circuits"):
             sample.target_probabilities = self._simulate_sample(sample)
 
         records = []
         for sample in samples:
             target_probabilities = {
-                f"p({key})": value for key, value in sample.target_probabilities.items()
+                f"p({key})": value
+                for key, value in sample.target_probabilities.items()  # type: ignore[union-attr]
```

Comment:
is it possible to avoid the type: ignore pragmas? e.g. with
```suggestion
                for key, value in sample.target_probabilities.items() if sample.target_probabilities is not None
```
or by combining this w/ the previous loop

Reply:
The logic meant that this case was never actually reached, hence the type: ignore. But adding the `if _ is not None` works much nicer!